elasticsearch/utils/es_util

```diff
@@ -20,14 +20,18 @@
 set -euo pipefail
 source es_util_env
 
+extra_args=("")
+index=0
+
 while (($#))
 do
 case $1 in
     --query=*)
       QUERY=${1#*=}
       ;;
     *)
-      echo Ignoring unknown argument $1
```

Comment:
how about just `break` here, so the remaining args are left in `"$@"`?

Reply:
or - have a case `--*) echo Error: unknown argument $1 ; exit 1 ;;`
then `*) break ;;`

Reply:
What if we decide to add another switch?

Reply:
@jcantrill just add it below `--query` and above `--*)`.  This assumes that all of the `--name` arguments will come before the other positional arguments.

If we think we are going to need more/better processing, then we should use `getopt` or something like that, rather than "roll-your-own" argument processing.